### PR TITLE
Namecheap dynamic DNS client

### PR DESCRIPTION
**Pull request by Huizerd**
Dynamic DNS client for Namecheap domains. Approach largely copied from the existing freeDNS script, adjusted according to [this gist](https://gist.github.com/ndunks/c756030c0757b667c9a478c97ca5a9b7) and [Namecheap docs](https://www.namecheap.com/support/knowledgebase/article.aspx/29/11/how-do-i-use-a-browser-to-dynamically-update-the-hosts-ip). Couldn't reopen #1184 due to force push.

### CODE REVIEW

File: etc/ncp-config.d/namecheapDNS.cfg [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      "id": "ACTIVE",
+      "name": "Active",
+      "value": "no",
+      "type": "bool"
+    },
+    {
+      "id": "PASSWORD",
+      "name": "Dynamic DNS password",
+      "value": "",
+      "suggest": "840deed9ef1844eca7c198df2d31bf10"
```
Comment (nachoparker) [inline etc/ncp-config.d/namecheapDNS.cfg:19]:
```suggestion
      "suggest": "your_password"
```

also this should be "type: password"

**Review by nachoparker — COMMENTED** [review]
thanks! looks good

**Comment by Huizerd** [comment]
Hey, does this need any more work? Would like to connect my Namecheap domain out of the box :)

**Comment by nachoparker** [comment]
great, thank you for your contribution

**Comment by Huizerd** [comment]
Hey, it says this was merged, but I can't find it in the devel or master branch? Should I add something?

**Comment by nachoparker** [comment]
bruh I can't find it either, I must have ran over it by mistake. I'll add it again

**Comment by nachoparker** [comment]
readded to devel

https://github.com/nextcloud/nextcloudpi/commit/24b60186a2e17a2f270ebda7c5eec53fe825633d

**Pull request merged**